core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -535,11 +535,17 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
             }
           } catch {
             case _: NoSuchElementException =>
-              // If the file is currently not being tracked by the SHS, add an entry for it and try
-              // to parse it. This will allow the cleaner code to detect the file as stale later on
-              // if it was not possible to parse it.
+              // If the file is currently not being tracked by the SHS, check whether the log file
+              // has expired, if expired, delete it from log dir, if not, add an entry for it and
+              // try to parse it. This will allow the cleaner code to detect the file as stale
+              // later on if it was not possible to parse it.
               try {
-                if (count < conf.get(UPDATE_BATCHSIZE)) {
+                if (conf.get(CLEANER_ENABLED) && reader.modificationTime <
+                  clock.getTimeMillis() - conf.get(MAX_LOG_AGE_S) * 1000) {
```

Comment:
@shuyouZZ , it seems that you reverted the indendation correction. Please fix the indentation here.

Reply:
What I'm asking is the following. In this PR, the line length of the next `logInfo` line is longer than the revised one. Please fix it.
```scala
 if (conf.get(CLEANER_ENABLED) && reader.modificationTime <
-  clock.getTimeMillis() - conf.get(MAX_LOG_AGE_S) * 1000) {
+    clock.getTimeMillis() - conf.get(MAX_LOG_AGE_S) * 1000) {
 logInfo(s"Deleting expired event log ${reader.rootPath.toString}")
```